src/Sylius/Bundle/AdminBundle/Resources/config/services.xml

```diff
@@ -25,6 +25,11 @@
             <tag name="sylius.context.locale" priority="128" />
         </service>
 
+        <service id="sylius.security.shop_user_impersonator" class="Sylius\Bundle\CoreBundle\Security\UserImpersonator" public="false">
+            <argument type="service" id="session" />
+            <argument type="string">%sylius_shop.firewall_context_name%</argument>
```

Comment:
What if we could just have a similar parameter in admin, and override its value if shop bundle is enabled? Otherwise, we would use some default value.